examples/gno.land/r/gov/dao/dao.gno

```diff
@@ -9,29 +9,33 @@ import (
 )
 
 var proposals = make([]*proposal, 0)
-
-// XXX var members ...
+var members = make([]std.Address, 0)
 
 type proposal struct {
-	idx      int
 	author   std.Address
 	comment  string
 	executor pproposal.Executor
-	// XXX: make "accepted" and "finished" managed by an interface that can have various voting implementations
-	accepted bool
-	finished bool
+	voter    Voter
 }
 
 func (p proposal) Status() Status {
+	if p.executor.IsExpired() {
+		return Expired
```

Comment:
I like that you've shifted the responsibility of the status from the executor to the Proposal, I think they overlapped before